files/k8s-resource/calico-all.yaml

```diff
@@ -309,6 +309,11 @@ spec:
         # priority scheduling and that its resources are reserved
         # if it ever gets evicted.
         scheduler.alpha.kubernetes.io/critical-pod: ''
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "9091"
+        giantswarm.io/monitoring: "true"
+        giantswarm.io/monitoring-path: "/metrics"
```

Comment:
IIRC this shouldn't start with a `/`, I'll double-check later, but see [monitoring of apps](https://intranet.giantswarm.io/docs/monitoring/monitoring_of_apps/).

Reply:
I think we need to adjust here as prometheus is scraping according to annotations only:
https://www.weave.works/docs/cloud/latest/tasks/monitor/configuration-k8s/#per-pod-prometheus-annotations Also, the path starts with a /

Reply:
Yeah _Prometheus_ itself is only using annotations, because internally it just gets a list of all resources and then filters it and applies relabelings to drop some according to expressions, etc. to narrow it down further.

We can't do that because we're using a selector to only get a subset of resources, and with selector you need a label, won't work with annotations.

As for the path, I suppose we can adjust our replacement expression so it expects the `/`, the only thing is I don't know how many managed apps we have already labelled without it and so how many we'd need to change.

Reply:
Alright, let me change it to a label. I am not sure they added this annotations in a lot of places though but maybe we can support both schemes?

Reply:
Do you mean to have `giantswarm.io/monitoring` as both annotation and a label? That could work too, and for apps it would mean they work with the current `g8s-prometheus` and our new _TC Prometheus_.

But for system components like calico, kube-proxy, etc. it would be good to test that this doesn't break the current `g8s-prometheus` setup, I have a notion it may cause errors because they would be picked up by two scrape configs - the custom workload one that matches system components, as well as the managed apps one.

Reply:
I was talking about the /metrics path, sorry for the confusion :)

Reply:
Yes, we still need to test it first. I will try tomorrow morning but testing k8scc is a pain

Reply:
Oh, right, with regards to monitoring path, it should work either way cause we rewrite that to `/api/v1/namespaces/${NS}/pods/${POD}:${PORT}/proxy/${PATH}` so if PATH starts with `/` it will just end up being doubled, e.g. `/api/v1/namespaces/${NS}/pods/${POD}:${PORT}/proxy//metrics`, which still works, it's just not as neat.

Reply:
Actually I also just realised that we don't need this at all here because `/metrics` is the default path.

Reply:
Then we just need to test it now :)